src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -281,19 +285,60 @@ func (r *Cluster) validatePandaproxyListeners() field.ErrorList {
 	return allErrs
 }
 
+func (r *Cluster) validateCPU() field.ErrorList {
+	var allErrs field.ErrorList
+
+	// CPU limit (if set) cannot be lower than the requested
+	if !r.Spec.Configuration.DeveloperMode &&
+		!r.Spec.Resources.Requests.Cpu().IsZero() && !r.Spec.Resources.Limits.Cpu().IsZero() &&
+		r.Spec.Resources.Limits.Cpu().Cmp(*r.Spec.Resources.Requests.Cpu()) == -1 {
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("spec").Child("resources").Child("requests").Child("cpu"),
+				r.Spec.Resources.Requests.Cpu(),
+				"CPU limit cannot be lower than the request, either increase the limit or remove it"))
+	}
+
+	return allErrs
+}
+
 // validateMemory verifies that memory limits are aligned with the minimal requirement of redpanda
-// which is 1GB per core
-// to verify this, we need to subtract the 1M we reserve currently for other processes
+// which is 1GB per core. To verify this, we need to subtract the 1M we reserve currently for other processes
 func (r *Cluster) validateMemory() field.ErrorList {
 	var allErrs field.ErrorList
+
+	// Ensure spare memory for other processes
 	quantity := resource.MustParse(ReserveMemoryString)
-	if !r.Spec.Configuration.DeveloperMode && (r.Spec.Resources.Limits.Memory().Value()-quantity.Value()) < gb {
+	if !r.Spec.Configuration.DeveloperMode && (r.Spec.Resources.Requests.Memory().Value()-quantity.Value()) < gb {
 		allErrs = append(allErrs,
 			field.Invalid(
-				field.NewPath("spec").Child("resources").Child("limits").Child("memory"),
+				field.NewPath("spec").Child("resources").Child("requests").Child("memory"),
 				r.Spec.Resources.Limits.Memory(),
-				"need minimum of 1GB + 1MB of memory per node"))
+				"need minimum request of 1GB + 1MB of memory"))
+	}
+
+	// Ensure a requested 2GB of memory per core
+	requests := r.Spec.Resources.Requests.DeepCopy()
+	requests.Cpu().RoundUp(0)
+	requestedCores := requests.Cpu().Value()
+	if !r.Spec.Configuration.DeveloperMode && r.Spec.Resources.Requests.Memory().Value() < requestedCores*MinimumMemoryPerCore {
```

Comment:
@senior7515 it's possible to run with 1GiB/core - should this allow 1GiB/core or require 2GiB/core?

Reply:
It is possible. we can downgrade to 1G/core, but not recommended for prod. It's very little memory - around 500MB free for request flow. 

I'm flexible i think we need to change the 'check' in `syschecks.h`

Reply:
The memory check I see is for a minimum of 1GiB (https://github.com/vectorizedio/redpanda/blob/dev/src/v/syschecks/syschecks.cc#L40)

Do we want to change that to 2GiB and then require the same through the operator, or the other way round (1GiB everywhere)?

Reply:
i think 2G/core is recommended. We'd need to test the alternative to have a different recommendation.

Reply:
We're bumping up against what a good default is vs. allowing users to override specific things. If we had more escape hatches for allowing the user to override a specific setting, this would be less of a problem.

Reply:
I'm not convinced that shadowing the actual configuration params with slightly different implications is useful. Can we support this: https://github.com/vectorizedio/redpanda/pull/1388 in the operator?

I'm happy with this PR as it stands, and to make more direct configuration possible in another PR.

Reply:
> Can we support this: #1388 in the operator?

I just sent a PR with a simpler version of 1388: https://github.com/vectorizedio/redpanda/pull/1491